backoff.go

```diff
@@ -0,0 +1,68 @@
+package deferred
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	maxDelay = 1000 * time.Second
+	factor   = 1.6
+	maxInt64 = float64(math.MaxInt64 - 512)
+)
+
+// Strategy defines the algorithm for backoff
+type Strategy interface {
```

Comment:
Does this need to be exported by the package, i.e. could it just be `strategy`? The same goes for the methods defined.